workflow/executor/docker/docker.go

```diff
@@ -185,6 +185,8 @@ func (d *DockerExecutor) GetExitCode(ctx context.Context, containerName string)
 }
 
 func (d *DockerExecutor) Wait(ctx context.Context, containerNames []string) error {
+	ctx, cancel := context.WithCancel(ctx) // stop the polling when we are no longer waiting
```

Comment:
It seems to be useless here to do `WithCancel(ctx)`. If the passed in `ctx` is cancelable, then there's no need to do it; If the passed in `ctx` is not cancelable, then I don't see a way could hit `ctx.Done()`.